http://bugs.gentoo.org/353816
http://sourceware.org/ml/libc-alpha/2011-02/msg00152.html

From 87fac357a6243982ce21cb71e21bbe0d35234eed Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Tue, 22 Feb 2011 13:26:19 -0500
Subject: [PATCH] memset: fix define usage for shared libs

The proper define to check "am I in a shared lib" is "SHARED", not "PIC".
The two new memset_chk functions incorrectly depend on "PIC".

Signed-off-by: Mike Frysinger <vapier@gentoo.org>

2011-02-22  Mike Frysinger  <vapier@gentoo.org>

	* sysdeps/i386/i686/memset_chk.S: Change PIC to SHARED.
	* sysdeps/x86_64/memset_chk.S: Likewise.
---
 sysdeps/i386/i686/memset_chk.S |    2 +-
 sysdeps/x86_64/memset_chk.S    |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/i386/i686/memset_chk.S b/sysdeps/i386/i686/memset_chk.S
index cd93d5e..e7511a0 100644
--- a/sysdeps/i386/i686/memset_chk.S
+++ b/sysdeps/i386/i686/memset_chk.S
@@ -20,7 +20,7 @@
 #include <sysdep.h>
 #include "asm-syntax.h"
 
-#ifndef PIC
+#ifndef SHARED
 	/* For libc.so this is defined in memset.S.
 	   For libc.a, this is a separate source to avoid
 	   memset bringing in __chk_fail and all routines
diff --git a/sysdeps/x86_64/memset_chk.S b/sysdeps/x86_64/memset_chk.S
index c1c8c23..2c4fffc 100644
--- a/sysdeps/x86_64/memset_chk.S
+++ b/sysdeps/x86_64/memset_chk.S
@@ -20,7 +20,7 @@
 #include <sysdep.h>
 #include "asm-syntax.h"
 
-#ifndef PIC
+#ifndef SHARED
 	/* For libc.so this is defined in memset.S.
 	   For libc.a, this is a separate source to avoid
 	   memset bringing in __chk_fail and all routines
-- 
1.7.4.1

